src/excmds.ts

```diff
@@ -3778,6 +3778,28 @@ export function seturl(pattern: string, key: string, ...values: string[]) {
     return config.setURL(pattern, ...validateSetArgs(key, values))
 }
 
+/**
+ * Usage: `setmode mode key values`
+ *
+ * @param mode The Mode the setting should be set for, e.g. `insert` or `ignore`.
+ * @param key The name of the setting you want to set, e.g. `allowautofocus`
+ * @param values The value you wish for, e.g. `true`
+ *
+ * Currently this command is only supported for the following settings:
+ * - [[allowautofocus]]
+ *
+ * Example:
+ * - `setmode ignore allowautofocus true`
+ */
+//#content
+export function setmode(mode: string, key: string, ...values: string[]) {
+    if (!mode || !key || !values.length) {
```

Comment:
It would be nice to do some check on mode being a valid ModeName, but I don't know how to do that, is there a way you can ask typescript for all the possible values for a type like that?

Reply:
Yeah it's something like `KeyOf TypeOf [mode object]` but it would get very messy quickly as users can make their own modes by just creating binds.

Reply:
Ah, I didn't know that, I'll just leave things as is then for now